src/btree/bt_sync.c

```diff
@@ -553,8 +559,12 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
             if (walk == NULL)
                 break;
 
-            /* Traverse through the internal page for obsolete child pages. */
-            if (F_ISSET(walk, WT_REF_FLAG_INTERNAL)) {
+            /*
+             * Perform checkpoint cleanup when not in startup or shutdown phase by traverse through
+             * the internal page for obsolete child pages.
```

Comment:
`traverse through the internal page for obsolete child pages`
should be
`traversing through internal pages looking for obsolete child pages`

Reply:
Fixed.